src/imitation/algorithms/preference_comparisons.py

```diff
@@ -1189,35 +1351,57 @@ def __init__(
                 on the fly by specifying an `epoch_multiplier` in `self.train()`
                 if longer training is desired in specific cases).
             lr: the learning rate
-            weight_decay: the weight decay factor for the reward model's weights
-                to use with ``th.optim.AdamW``. This is similar to but not equivalent
-                to L2 regularization, see https://arxiv.org/abs/1711.05101
-            seed: seed for the internal RNG used in bagging
+            seed: the random seed to use for splitting the dataset into training
```

Comment:
`seed` and `rng_seed` are rather confusing names. Aren't all seeds used to seed random number generators (RNGs)? `seed` also suggests some more primacy or importance ove the longer name `rng_seed`, but they seem similarly important to me?

I'd give them clearer names: `bagging_seed` and `split_seed`, for example.

Is there a reason to even separate them? Why not just use the same seed for both?

Reply:
Lol yes when I made this change I didn't want to touch stuff that someone else had added (I merged incoming changes from master) so I thought that would be an acceptable fix. Actually thinking about what the code is doing, having the same seed for both is probably more sensible.

Even though the mypy PR is trying to replace everything with RandomStates directly, but since the point of the PR is to improve type hints, I think it's fine if there are some inconsistencies still, those should be addressed and cleaned up on a separate PR.

Reply:
Actually the reason why I did this is because they have different defaults (0 and None), and that's because Generator.manual_seed seems to require an integer. But I can fix this by not passing a generator if no seed is passed.